apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,672 @@
+  end,
+  {next_state, zero, State}.
+
+pools_(start, #state{pools = [_]} = State) ->
+  {next_state, pools, State};
+
+pools_(start, #state{} = State) ->
+  {next_state, pools, send_check(State)};
+
+pools_(check, #state{pools = [_]} = State) ->
+  {next_state, pools, State};
+
+pools_(check, #state{pools = [{Pool, _}|_], node_to_id = NodeToId} = State) ->
+  stop_pool(Pool, NodeToId),
+  {next_state, pools, send_check(State)};
+
+pools_({pool_connected, Node}, #state{pools = [Pool], passive_pools = PassivePools} = State) ->
+  {next_state, pools, send_check(State#state{pools = connect(Node, [Pool]), passive_pools = lists:delete(Node, PassivePools)})};
+
+pools_({pool_connected, Node}, #state{pools = Pools, passive_pools = PassivePools} = State) ->
+  {next_state, pools, State#state{pools = connect(Node, Pools), passive_pools = lists:delete(Node, PassivePools)}};
+
+pools_({pool_enabled, Node}, #state{pools = Pools} = State) ->
+  {next_state, pools, State#state{pools = enable(Node, Pools)}};
+
+pools_({task_accepted, Ref, Node}, #state{pools = Pools} = State) ->
+  NewPools =
+    case status(Node, Pools) of
+      {pending, Ref, _Id} ->
+        enable(Node, Pools);
+      _ ->
+        Pools
+    end,
+  {next_state, pools, State#state{pools = NewPools}};
+
+pools_({pool_disabled, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  NewPools = disable(Node, Pools),
+  {NextState, NewState} =
+  case status(Node, Pools) of
+    {pending, _Ref, Id} ->
+      case task(Category, Id) of
+        undefined ->
+          case is_enabled(NewPools) of
+            true -> {pools, State#state{pools = NewPools}};
+            false -> {zero, State#state{pools = NewPools}}
+          end;
+        Task ->
+          rpc:cast(Node, fyler_pool, cancel_task, [Id]),
+          choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = NewPools})
+      end;
+    enabled ->
+        case is_enabled(NewPools) of
+          true -> {pools, State#state{pools = NewPools}};
+          false -> {zero, State#state{pools = NewPools}}
+        end
+  end,
+  {next_state, NextState, NewState};
+
+pools_({pool_down, Node}, #state{category = Category, retasks = ReTasks, pools = Pools, passive_pools = PassivePools} = State) ->
+  NewReTasks = restart_tasks(Category, ReTasks, Node),
+  {NextState, NewState} = choose_pool(State#state{retasks = NewReTasks, pools = delete(Node, Pools), passive_pools = [Node|PassivePools]}),
+  {next_state, NextState, NewState};
+
+pools_({task_rejected, Ref, Node}, #state{category = Category, retasks = ReTasks, pools = Pools} = State) ->
+  case status(Node, Pools) of
+    {pending, Ref, Id} ->
+      NewPools = enable(Node, Pools),
+      case task(Category, Id) of
+        undefined ->
+          {next_state, pools, State#state{pools = NewPools}};
+        Task ->
+          {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = enable(Node, Pools)}),
```

Comment:
Line 318 is too long:           {NextState, NewState} = choose_pool(State#state{retasks = fyler_queue:in(Task, ReTasks), pools = enable(Node, Pools)}),.